atc/db/migration/migrations/1597678793_add_job_id_to_build_image_resource_caches.up.sql

```diff
@@ -0,0 +1,12 @@
+BEGIN;
+  ALTER TABLE build_image_resource_caches
+    ADD COLUMN job_id integer REFERENCES jobs(id);
```

Comment:
This is the controversial part of this change, since it denormalizes (there is redundancy between `job_id` on `build_image_resource_caches` and the result of joining `builds` on `builds.id = build_image_resource_caches.build_id` and selecting `builds.job_id`). I have to wonder if there's a better way.

Reply:
Right I wanted to remove the need to join to the `builds` table in order to fetch the `job_id` from the `build` because that is the part that makes the current query slow. This is definitely something I am open to finding a different way to not need to duplicate the `job_id` data everywhere because this is something I run into a lot, where the thing that makes a query slow is the joining to the `builds` table in order to grab the `job_id` from it.